### ISSUE TITLE
Aligning problem on XCode 8

### ISSUE THREAD

**Issue by gokhanakkurt** [ninjaprox/NVActivityIndicatorView#73]
I'm having trouble to align the indicator view on storyboard. I have a basic app and single view controller. Indicator is located to the center of view and it spins until the page contents load from the server. However, when I hit the run button, loading indicator is spinning in the right corner of the view, overflowing and not aligning to the center itself.

Scenario can be replicated by following steps:
- Add a `UIView` to a `UIViewController` on storyboard.
- Add static height and width and contraints (let say 100,100)
- Align center vertically and horizontally
- Change the class of `UIView` to `NVActivityIndicatorView`
- Assign activity type and color.
- Hit the run button.

I'm building with `XCode 8` and `Swift 3.0`. Is there any way to fix this? 

Thanks in advance.

**Comment by gokhanakkurt** [comment]
Creating an instance of `NVActivityIndicatorView` and aligning programmatically always works as you stated in your example. However, storyboard problem still exists.

``` swift
class TestViewController: UIViewController {

    private var activityView: NVActivityIndicatorView!

    override func viewDidLoad() {
        super.viewDidLoad()
        showLoadingIndicator()
    }

    func showLoadingIndicator(){
        if activityView == nil{
            activityView = NVActivityIndicatorView(frame: CGRect(x: 0, y: 0, width: 100.0, height: 100.0), type: NVActivityIndicatorType.ballClipRotate, color: UIColor.black, padding: 0.0)
            // add subview
            view.addSubview(activityView)
            // autoresizing mask
            activityView.translatesAutoresizingMaskIntoConstraints = false
            // constraints
            view.addConstraint(NSLayoutConstraint(item: activityView, attribute: NSLayoutAttribute.centerX, relatedBy: NSLayoutRelation.equal, toItem: view, attribute: NSLayoutAttribute.centerX, multiplier: 1, constant: 0))
            view.addConstraint(NSLayoutConstraint(item: activityView, attribute: NSLayoutAttribute.centerY, relatedBy: NSLayoutRelation.equal, toItem: view, attribute: NSLayoutAttribute.centerY, multiplier: 1, constant: 0))
        }

        activityView.startAnimating()
    }

    func stopLoadingIndicator(){
        activityView.stopAnimating()
    }
}
```

**Comment by siberianisaev** [comment]
Check the frame inside NVActivityIndicatorView.swift
`required public init?(coder aDecoder: NSCoder)`
i have the same issue, width/height constraints set to 100px for activity, but decoded view is 1000x1000px, looks like storyboard bug.

My problem persist only for one screen (split view detail controller), on other (not related to split view) screens all is ok

**Comment by siberianisaev** [comment]
@gokhanakkurt as workaround try to wrap your activity with _two_ other views inside storyboard (containers with the same size).

**Comment by gokhanakkurt** [comment]
I've achieved by using storyboard too. However, I don't understand why this is happening.

I placed the view on storyboard and assigned to center both vertically and horizontally constraints. Then I created an `IBOutlet` on `UIViewController`.  I invoke `startAnimating()` function on `viewDidAppear(_:)` as follow:

``` swift
override func viewDidAppear(_ animated: Bool) {
   super.viewDidAppear(animated)
   activityView.startAnimating()
}
```

It works but not expected one!

**Comment by ninjaprox** [comment]
I also encounter this issue. Here is my simple check.

``` swift
import UIKit
import Foundation
import NVActivityIndicatorView

class ViewController: UIViewController {
    @IBOutlet weak var activityIndicator: NVActivityIndicatorView!
    @IBOutlet weak var normalView: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()
        activityIndicator.startAnimating()
        print(normalView.frame)
        print(activityIndicator.frame)
    }
}
```

Output

```
(0.0, 0.0, 1000.0, 1000.0)
(0.0, 0.0, 1000.0, 1000.0)
```

Regardless different constraints of 2 views, their frame are always 1000x1000. Any idea why this's happening? Is there a change on layout or `UIViewController` life cycle on iOS 10?

**Comment by gokhanakkurt** [comment]
@ninjaprox I'm thinking of the same thing. I'll look at the lifecycle of `UIViewController`.

**Comment by gokhanakkurt** [comment]
@ninjaprox I haven't seen any change in lifecycle.  Any update?

**Comment by siberianisaev** [comment]
Interesting it's content size ![content size](https://dl.dropboxusercontent.com/u/25357310/NVActivityIndicatorView_issue73.png)

**Comment by gokhanakkurt** [comment]
@siberianisaev just saw [this](http://stackoverflow.com/a/39541653/1129468) and [that](http://stackoverflow.com/a/39589860/1129468) on `StackOverflow`.  These could clarify the issue.

**Comment by siberianisaev** [comment]
Workaround:
1. create **container** for activity in storyboard
2. then add activity in **viewDidLayoutSubviews**

`fileprivate weak var activity: NVActivityIndicatorView?
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        
        addActivity()
    }
    
    fileprivate func addActivity() {
        if nil == activity {
            let frame = container.bounds
            let a = NVActivityIndicatorView(frame: frame, type: .ballScaleMultiple, color: UIColor.red, padding: 0)
            container.addSubview(a)
            
            activity = a
        }
    }`

**Comment by siberianisaev** [comment]
``` swift
@IBOutlet weak var activity: NVActivityIndicatorView!
@IBOutlet weak var activityHeight: NSLayoutConstraint!
```

in -viewDidLoad just reset frame

``` swift
var frame = activity.frame
let c = activityHeight.constant
frame.size = CGSize(width: c, height: c)
activity.frame = frame
```

**Comment by gokhanakkurt** [comment]
@siberianisaev Your suggestions are OK but we should keep digging into the problem to find the root cause. If it's bug of `XCode 8`, it's not necessary to create workarounds here. The only thing we need is to wait for next release for `XCode`.

**Comment by siberianisaev** [comment]
Constant 1000x1000 size suggests the idea that interface object is not properly decoded by Xcode 8. Probably width/height replaced by value of some constraint priority (@ 1000).

**Comment by ninjaprox** [comment]
I think this is mostly bug in Xcode 8. Workaround may be needed in the meantime waiting for Xcode update.

**Comment by pilot34** [comment]
I overrode `NVActivityIndicatorView` and added such hack. That helped.

``` swift
    override func awakeFromNib() {
        super.awakeFromNib()

        // Xcode 8 workaround
        if self.frame.size.width == 1000 && self.frame.size.height == 1000 {
            guard let widthConstraint = (self.constraints.filter { return $0.firstAttribute == .Width }).first else { return }
            guard let heightConstraint = (self.constraints.filter { return $0.firstAttribute == .Height }).first else { return }
            self.frame.size = CGSizeMake(widthConstraint.constant, heightConstraint.constant)
        }
    }
```

**Comment by siberianisaev** [comment]
@pilot34 missed cases with using `aspect` constraint + (`width` OR `height` constraint).

**Comment by siberianisaev** [comment]
Similar issues:
- https://forums.developer.apple.com/thread/63489
- https://openradar.appspot.com/28325425
- http://stackoverflow.com/questions/39578530/since-xcode-8-and-ios10-views-are-not-sized-properly-on-viewdidlayoutsubviews

**Comment by jitaek** [comment]
Hopefully we get a proper fix, but simply adding `layoutIfNeeded()` to my `UITableViewCell`'s `awakeFromNib` function fixed this problem.

**Comment by v-silin** [comment]
If this still actual issue for someone, I fixed it by calling updateConstraints and layoutIfNeeded before start animations.

**Comment by ninjaprox** [comment]
This has been fixed in Xcode 8.1 🎉.
